src/backend/aspen/database/models/sample.py

```diff
@@ -46,6 +48,29 @@ class RegionType(enum.Enum):
 )
 
 
+def create_public_id(context) -> str:
+    current_parameters: Mapping[
+        str, Union[str, bool, datetime]
+    ] = context.get_current_parameters()
+    interface = SqlAlchemyInterface(context.engine)
+    with session_scope(interface) as session:
+        group: Group = (
+            session.query(Group)
+            .filter(Group.id == current_parameters["submitting_group_id"])
+            .one()
+        )
+        all_samples: Collection[Sample] = session.query(Sample).all()
```

Comment:
This is going to be pretty computationally expensive if we're calling this on every new sample we make. Could we do some pre calculating of how many ids we're going to need to generate outside of the for loop in the post submission and then generate them in bulk?

Reply:
ACTUALLY scratch that, looks like sqlalchmey has a max id functionality https://stackoverflow.com/questions/3750118/sqlalchemy-0-5-8-max-function 
let's use this instead of running this query.

Reply:
oh nice!!

Reply:
added!